### PR TITLE
Run acceptance and unit tests when test fixtures change

### PR DESCRIPTION
**Pull request by trodge**
fixes https://github.com/hashicorp/terraform-provider-google/issues/17906

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/magician/cmd/test_terraform_vcr.go [hunk 1]
```diff
@@ -292,7 +292,7 @@ func modifiedPackages(newBranch, oldBranch string, rnr ExecRunner) (map[string]s
 	}
 	var goFiles []string
```
Comment (melinath) [inline .ci/magician/cmd/test_terraform_vcr.go:293]: Could you split the logic from here to the end of the function into a separate function and add thorough unit tests for it?

File: .ci/magician/cmd/test_terraform_vcr.go [hunk 2]
```diff
@@ -292,7 +292,7 @@ func modifiedPackages(newBranch, oldBranch string, rnr ExecRunner) (map[string]s
...
 	for _, line := range strings.Split(diffs, "\n") {
-		if strings.HasSuffix(line, ".go") || line == "go.mod" || line == "go.sum" {
+		if strings.HasSuffix(line, ".go") || strings.Contains(line, "test-fixtures") || line == "go.mod" || line == "go.sum" {
```
Comment (melinath) [inline .ci/magician/cmd/test_terraform_vcr.go:295]: I see that the bash version of this is matching against `go.mod$` - why is that? Should we use HasSuffix here?

**Review by melinath — CHANGES_REQUESTED** [review]
makes sense - one refactor to be able to test better

File: .ci/magician/cmd/test_terraform_vcr.go (hunk 2 above)
Comment (trodge) [reply .ci/magician/cmd/test_terraform_vcr.go:295]: I don't think there's any particular reason other than that the other pattern ends in `$`. We don't have any other files that end with `go.mod` or `go.sum`.

Comment (trodge) [reply .ci/magician/cmd/test_terraform_vcr.go:299]: Oh, never mind. I forgot the line contains the full path of the file.

File: .ci/magician/cmd/test_terraform_vcr_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package cmd
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestModifiedPackagesFromDiffs(t *testing.T) {
```
Comment (melinath) [inline .ci/magician/cmd/test_terraform_vcr_test.go:8]: could you add `go.mod` and `go.sum` cases, and some other cases that shouldn't trigger tests (such as markdown changes, or hypothetical changes to other files in google-beta that wouldn't match?)

**Review by melinath — APPROVED** [review]

**Review by melinath — APPROVED** [review]

**Pull request merged**